src/useChannelClient.ts

```diff
@@ -0,0 +1,29 @@
+import { ChannelClient } from "openfin/_v2/api/interappbus/channel/client";
+import { useEffect, useState } from "react";
+import { ChannelAction } from "./types/ChannelAction";
+
+export default (channelName: string, pushActions?: ChannelAction[]) => {
```

Comment:
Do you think it would be nice if the push actions changed to register/unregister them in another `useEffect`?

Reply:
I originally thought I would define the pushActions prior to creating the ChannelClient, but like you -- I found it easier to have the ChannelsClient and then register the action. I've removed this from the API.